src/main/java/org/broadinstitute/hellbender/tools/copynumber/utils/annotatedregion/SimpleAnnotatedGenomicRegionCollection.java

```diff
@@ -0,0 +1,261 @@
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.codecs.xsvLocatableTable.XsvLocatableTableCodec;
+import org.broadinstitute.hellbender.utils.codecs.xsvLocatableTable.XsvTableFeature;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.IntStream;
+
+/**
+ * Represents a collection of annotated regions.  The annotations do not need to be known ahead of time, if reading from a file.
+ * Though {@link SimpleAnnotatedGenomicRegion::CONTIG_HEADER},
+ *  {@link SimpleAnnotatedGenomicRegion::START_HEADER}, and
+ *  {@link SimpleAnnotatedGenomicRegion::END_HEADER} are expected, by default, for defining the genomic region.
+ *
+ *  This class supports reading xsv (tsv, csv) files with comments ("#") and SAM headers ("@").  The default is tsv.
+ */
+public class SimpleAnnotatedGenomicRegionCollection {
+
+    static final Logger logger = LogManager.getLogger(SimpleAnnotatedGenomicRegionCollection.class);
+
+    public static final String ANNOTATED_REGION_DEFAULT_CONFIG_RESOURCE = "org/broadinstitute/hellbender/tools/copynumber/utils/annotatedregion/annotated_region_default.config";
+    private SAMFileHeader samFileHeader;
+
+    /** Does not include the locatable fields. */
+    private List<String> annotations;
+    private List<String> comments;
+    private List<SimpleAnnotatedGenomicRegion> records;
+
+    /* The output contig column name */
+    private String contigColumnName;
+
+    /* The output start column name */
+    private String startColumnName;
+
+    /* The output end column name */
+    private String endColumnName;
+
+    private SimpleAnnotatedGenomicRegionCollection(final SAMFileHeader samFileHeader, final List<String> annotations, final List<String> comments,
+                                                   final List<SimpleAnnotatedGenomicRegion> records, final String contigColumnName,
+                                                   final String startColumnName, final String endColumnName) {
+        this.comments = comments;
+        this.samFileHeader = samFileHeader;
+        this.annotations = annotations;
+        this.records = records;
+        this.contigColumnName = contigColumnName;
+        this.startColumnName = startColumnName;
+        this.endColumnName = endColumnName;
+    }
+
+    /**
+     *  Same as {@link #create(Path, Path, Set)} , but uses the default annotation
+     *   region config file in the GATK.
+     * @param input See {@link #create(Path, Path, Set)}
+     * @param headersOfInterest See {@link #create(Path, Path, Set)}
+     * @return See {@link #create(Path, Path, Set)}
+     */
+    public static SimpleAnnotatedGenomicRegionCollection create(final Path input, final Set<String> headersOfInterest) {
```

Comment:
Another thought: some of the confusion here comes from the desired functionality of being able to read either `columnsOfInterest` or all columns.

Why not just always read all available columns in the file?  You'd only need a single relatively simple read method that knows how to dynamically parse the column header.

Then you could have a `subsetAnnotations` method in the record class that takes `columnsOfInterest` (although I probably wouldn't name the parameter as such) and returns a new record with the specified annotations (and throws an exception if an unavailable annotation is requested).  

You could have a corresponding method in the collections class, which would just be a one-liner with streams.

Reply:
I have reduced the number of create methods.

I also made the create method that takes in a config file private.  This will enforce the input files having standardized contig, start, and end columns.

Reply:
Done (with your approval)